### PR TITLE
IDE projects updated with http library changes

### PR DESCRIPTION
**Pull request by leegeth**
IDE projects updated with http library changes

Description
-----------
IDE project files with http library changes

Checklist:
----------
- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/infineon/xmc4800_iotkit/dave4/aws_demos/.project [hunk 1]
```diff
@@ -653,6 +653,11 @@
 			<type>1</type>
 			<locationURI>AFR_HOME/libraries/freertos_plus/standard/pkcs11/src/iot_pkcs11.c</locationURI>
 		</link>
+		<link>
```
Comment (sarenameas) [inline projects/infineon/xmc4800_iotkit/dave4/aws_demos/.project]: Not sure why the iot_pkcs11_pal.c is moved, but we shall leave it and check it in.

**Comment by sarenameas** [comment]
Just fixed the build error in MediaTek.

File: projects/nxp/lpc54018iotmodule/iar/aws_demos/aws_demos.ewp [hunk 2]
```diff
@@ -1088,7 +1090,7 @@
 		</settings>
 	</configuration>
 	<group>
-		<name>freertos_kernel</name>
+		<name>freertos_kernel/</name>
```
Comment (sarenameas) [inline projects/nxp/lpc54018iotmodule/iar/aws_demos/aws_demos.ewp]: This is weird

File: projects/nxp/lpc54018iotmodule/iar/aws_demos/aws_demos.ewp [hunk 3]
```diff
@@ -1107,1874 +1109,1757 @@
...
-							<name>$PROJ_DIR$\..\..\..\..\..\libraries\3rdparty\mbedtls\include\mbedtls\error.h</name>
-						</file>
-						<file>
-							<name>$PROJ_DIR$\..\..\..\..\..\libraries\3rdparty\mbedtls\include\mbedtls\gcm.h</name>
-						</file>
-						<file>
-							<name>$PROJ_DIR$\..\..\..\..\..\libraries\3rdparty\mbedtls\include\mbedtls\havege.h</name>
-						</file>
-						<file>
-							<name>$PROJ_DIR$\..\..\..\..\..\libraries\3
```
Comment (sarenameas) [inline projects/nxp/lpc54018iotmodule/iar/aws_demos/aws_demos.ewp]: Not sure if this is OK

Comment (sarenameas) [inline projects/nxp/lpc54018iotmodule/iar/aws_demos/aws_demos.ewp]: Yea looks like the whole tree is flattening.

File: projects/nxp/lpc54018iotmodule/iar/aws_tests/aws_tests.ewp [hunk 4]
```diff
@@ -1115,2024 +1119,1901 @@
...
-					<file>
-						<name>$PROJ_DIR$\..\..\..\..\..\libraries\3rdparty\mbedtls\utils\mbedtls_utils.c</name>
-					</file>
-				</group>
-				<group>
-					<name>include</name>
-					<group>
-						<name>mbedtls</name>
-						<file>
-							<name>$PROJ_DIR$\..\..\..\..\..\lib
```
Comment (sarenameas) [inline projects/nxp/lpc54018iotmodule/iar/aws_tests/aws_tests.ewp]: Yea the tests is flattened too

File: projects/pc/windows/visual_studio/aws_demos/aws_demos.vcxproj [hunk 5]
```diff
@@ -1,524 +1,451 @@
...
-    <ClCompile Include="..\..\..\..\..\freertos_kernel\tasks.c" />
-    <ClCompile Include="..\..\..\..\..\freertos_kernel\timers.c" />
-    <ClCompile Include="..\..\..\..\..\libraries\3rdparty\http-parser\http_parser.c" />
-    <ClCompile Include="..\..\..\..\..\libraries\3rdparty\jsmn\jsmn.c" />
-    <ClCompile Include="..\..\..\..\..\libraries\3rdparty\mbedtls\library\aes.c" />
-    <ClCompile Include="..\..\..\..\..\libraries\3rdparty\mbedtls\library\aesni.c" />
-    <ClCompile Include="..\..\..\..\..\libraries\3rdparty\mbedtls\library\arc4.c" />
-    <ClCompile Include="..\..\..\..\..\libraries\3rdparty\mbedtls\library\asn1parse.c" />
-    <ClCompile Include="..\..\..\..\..\libraries\3rdparty\mbedtls\library\asn1write.c">
-      <WarningLevel Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">TurnOffAllWarnings</WarningLevel>
```
Comment (sarenameas) [inline projects/pc/windows/visual_studio/aws_demos/aws_demos.vcxproj]: Not sure if these special warnings need to be carried over.

File: projects/pc/windows/visual_studio/aws_tests/aws_tests.vcxproj.filters [hunk 6]
```diff
@@ -1,1719 +1,1337 @@
...
-    <ClCompile Include="..\..\..\..\..\libraries\abstractions\platform\freertos\iot_metrics.c">
-      <Filter>libraries\abstractions\platform\freertos</Filter>
-    </ClCompile>
-    <ClCompile Include="..\..\..\..\..\libraries\abstractions\platform\freertos\iot_network_freertos.c">
-      <Filter>libraries\abstractions\platform\freertos</Filter>
-    </ClCompile>
-    <ClCompile Include="..\..\..\..\..\libraries\abstractions\platform\freertos\iot_threads_freertos.c">
-      <Filter>libraries\abstractions\platform\freertos</Filter>
-    </ClCompile>
-    <ClCompile Include="..\..\..\..\.
```
Comment (sarenameas) [inline projects/pc/windows/visual_studio/aws_tests/aws_tests.vcxproj.filters]: Supposedly aws_demos and aws_tests both look bad in the GUI. Need to look at it.

File: projects/pc/windows/visual_studio/aws_demos/aws_demos.vcxproj (hunk 5 above)
Comment (sarenameas) [reply projects/pc/windows/visual_studio/aws_demos/aws_demos.vcxproj]: Opened visual studio on this branch manually and everything is fine.

File: projects/pc/windows/visual_studio/aws_tests/aws_tests.vcxproj.filters (hunk 6 above)
Comment (sarenameas) [reply projects/pc/windows/visual_studio/aws_tests/aws_tests.vcxproj.filters]: Looked and ran the test project and the demo project and everything looks good.

**Comment by jainamt** [comment]
@abhidixi11 today pointed out that the Cypress project files are not updated. @leegeth can you please check if that got missed accidentally.

**Comment by sarenameas** [comment]
> @abhidixi11 today pointed out that the Cypress project files are not updated. @leegeth can you please check if that got missed accidentally.

We have discussed this offline. Eric is working on the update.

**Comment by jainamt** [comment]
> > @abhidixi11 today pointed out that the Cypress project files are not updated. @leegeth can you please check if that got missed accidentally.
> 
> We have discussed this offline. Eric is working on the update.

Thanks

File: vendors/cypress/boards/CYW943907AEVAL1F/aws_demos/config_files/aws_demo_config.h [hunk 7]
```diff
@@ -36,6 +36,8 @@
  *          CONFIG_DEFENDER_DEMO_ENABLED
  *          CONFIG_POSIX_DEMO_ENABLED
  *          CONFIG_OTA_UPDATE_DEMO_ENABLED
+ *          CONFIG_HTTPS_SYNC_DOWNLOAD_DEMO_ENABLED
+ *          CONFIG_HTTPS_ASYNC_DOWNLOAD_DEMO_ENABLED
```
Comment (sarenameas) [inline vendors/cypress/boards/CYW943907AEVAL1F/aws_demos/config_files/aws_demo_config.h]: Thanks @ericyeh1995 !

**Pull request merged**